diff --git a/CMakeLists.txt b/CMakeLists.txt
index 42d6876ef57..dd2dd48f612 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,7 +93,11 @@ endif ()
 # C++17 is needed by std::filesystem and parquet.
 # C++14 is needed by Google Test >= 1.13, for all the other parts C++11 should be enough.
 # This will silently fall back to C++14 or even C++11, whatever is supported by the compiler.
-set(CMAKE_CXX_STANDARD 17)
+if (${CMAKE_VERSION} VERSION_LESS 3.8.0)
+    set(CMAKE_CXX_STANDARD 14)
+else()
+    set(CMAKE_CXX_STANDARD 17)
+endif()
 if (NOT "cxx_std_17" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
     message(WARNING "Your compiler does not support C++17. Please consider upgrading your compiler toolchain.")
 endif()
