diff --git a/src/libsumo/TraCIDefs.h b/src/libsumo/TraCIDefs.h
index bf84c8c45e6..ae873052ce4 100644
--- a/src/libsumo/TraCIDefs.h
+++ b/src/libsumo/TraCIDefs.h
@@ -378,6 +378,7 @@ struct TraCILogic {
 
 
 struct TraCILink {
+    TraCILink() {}
     TraCILink(const std::string& _from, const std::string& _via, const std::string& _to)
         : fromLane(_from), viaLane(_via), toLane(_to) {}
     ~TraCILink() {}
